independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Injection.java

```diff
@@ -110,7 +112,7 @@ private static void forClassBean(ClassInfo beanClass, ClassInfo classInfo, BeanD
 
         for (DotName resourceAnnotation : beanDeployment.getResourceAnnotations()) {
             List<AnnotationInstance> resourceAnnotations = getAllInjectionPoints(beanDeployment, classInfo,
-                    resourceAnnotation);
+                    resourceAnnotation, true);
             if (resourceAnnotations != null) {
```

Comment:
Not introduced by this PR, but I think that `resourceAnnotations` is always not null.